src/poly/dcrt/matrix/dcrt_poly_matrix.rs

```diff
@@ -352,6 +322,24 @@ impl DCRTPolyMatrix {
         debug_mem(format!("GetMatrixElement row={}, col={}", nrow, ncol));
         DCRTPolyMatrix::from_poly_vec(params, matrix_inner)
     }
+
+    pub(crate) fn gen_dcrt_gadget_vector(params: &DCRTPolyParams) -> DCRTPolyMatrix {
+        let base = 1 << params.base_bits();
+        let g_vec_cpp = DCRTPolyGadgetVector(
+            params.ring_dimension(),
+            params.crt_depth(),
+            params.crt_bits(),
+            params.modulus_digits(),
+            base,
+        );
+        DCRTPolyMatrix::from_cpp_matrix_ptr(params, &CppMatrix::new(g_vec_cpp))
+    }
+
+    pub(crate) fn dcrt_decompose_poly(&self, poly: &DCRTPoly, base_bits: u32) -> Vec<DCRTPoly> {
```

Comment:
```suggestion
    pub(crate) fn dcrt_decompose_poly(poly: &DCRTPoly, base_bits: u32) -> Vec<DCRTPoly> {
```

Reply:
Can we add it as a `decompose` method to the polynomial?

Reply:
@enricobottazzi 
I think this function cannot be the same as the existing decompose function in `DCRTPoly` even when the base is 2.
While the function in `DCRTPoly` regards the polynomial coefficient as a big integer, the `dcrt_decompose_poly` function regards it as a vector of integers in the CRT format.
Since the latter function is used only in `dcrt_poly_matrix.rs`, I think we can keep this function (or make it a private function).

Reply:
Ok, I agree on making it as a private function!